src/components/DayPicker.jsx

```diff
@@ -695,7 +695,7 @@ class DayPicker extends React.PureComponent {
     }
 
     if (newMonth && (!focusedDate || !isDayVisible(focusedDate, newMonth, numberOfMonths))) {
-      focusedDate = newMonth.clone().startOf('month');
+      focusedDate = newMonth.clone().startOf('month').hour(12);
```

Comment:
similarly, something that unit tests or exercises getFocusedDay, and shows the bug that this change fixes, would cover here

Reply:
Bug is not related to timezone, it's related to initial time for newly created date objects because in some code lines hours are set while in others are not. So I'm in stuck what test is needed here.

Reply:
If this change isn't fixing anything, maybe it should be reverted?

Reply:
> If this change isn't fixing anything, maybe it should be reverted?

Sure, but there is a good reason to have unified approach everywhere: it minimizes possible FUTURE bugs. All developers will see how to get first day of month and in all future cases will get a date with proper initial time.

Only "hour(12)" in components really matter for the issue.